examples/cpp/raw_mandelbrot.cpp

```diff
@@ -0,0 +1,175 @@
+#include <stdlib.h>
+#include <dlfcn.h>
+#include <stdio.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <new>
+
+#include "pf.h"
+
+#include "model/colormap.h"
+#include "model/image.h"
+#include "model/enums.h"
+#include "model/imagewriter.h"
+
+#include "model/fractfunc.h"
+#include "model/vectors.h"
+
+
+int main() {
+    // initial setup: load fract4_stdlib globally so the loaded formula has access to it
+    void *fract_stdlib_handle = dlopen("./fract_stdlib.so", RTLD_GLOBAL | RTLD_NOW);
+    if (NULL == fract_stdlib_handle) {
+        fprintf(stderr, "Error loading libfract_stdlib: %s", dlerror());
+        return -1;
+    }
+
+    // load formula lib
+    void *lib_handle = dlopen("./formula.so", RTLD_NOW);
+    if (NULL == lib_handle)
+    {
+        fprintf(stderr, "Error loading formula: %s", dlerror());
+        return -1;
+    }
+    pf_obj *(*pfn)(void);
+    pfn = (pf_obj * (*)(void)) dlsym(lib_handle, "pf_new");
+    if (NULL == pfn)
+    {
+        fprintf(stderr, "Error loading formula symbols: %s", dlerror());
+        dlclose(lib_handle);
+        return -1;
+    }
+    pf_obj *pf_handle = pfn();
+
+    // formula params: [0, 4.0, 0.0, 1.0, 4.0, 0.0, 1.0]
+    int param_len = 7;
+    struct s_param *params = (struct s_param *)malloc(param_len * sizeof(struct s_param));
```

Comment:
`auto params{std::make_unique<s_param []>(param_len)};` for both automatic memory management, and non-UB behaviour. malloc() does not start object lifetimes, so every operation to params after this is undefined behaviour.
Only new/new [] can start an object's lifetime in C++.